pkg/webservice/state_dump.go

```diff
@@ -157,29 +161,52 @@ func doStateDump(w io.Writer) error {
 	if err != nil {
 		return err
 	}
-	if _, err = w.Write(prettyJSON); err != nil {
+
+	stateLog := log.New(w, "", 0)
+	stateDumpFilePath := getStateDumpFilePath(schedulerContext)
+	stateLog.SetOutput(&lumberjack.Logger{
+		Filename:   stateDumpFilePath,
+		MaxSize:    10,
+		MaxBackups: 10,
+	})
+
+	if err = stateLog.Output(stateLogCallDepth, string(prettyJSON)); err != nil {
 		return err
 	}
 
 	return nil
 }
 
+func getStateDumpFilePath(cc *scheduler.ClusterContext) string {
+	cc.RLock()
+	defer cc.RUnlock()
+
+	for _, partition := range cc.GetPartitionMapClone() {
+		if partition.GetStateDumpFilePath() != "" {
+			return partition.GetStateDumpFilePath()
+		}
```

Comment:
Since state dump returns info of all partitions, but the state dump output path is a partition-level config property, we iterate through partitions until we find first occurrence of the property. This first occurrence will be used as the output path if multiple values are configured for different partitions. If no occurrences, we default to `yunikorn-state.txt`.

Reply:
> Just have a question - this file path is independent of the partitions, so it feels more logical to place this in `SchedulerConfig` instead of `PartitionConfig`. I know @yangwwei recommended this as a partition-level setting, just wondering if lifting it one level further has more drawbacks.

Right now, we do not have a global "schedulerConfig" field, all configs are partition-based. It is not a big problem now because we only support a single partition by far.  I agree it is more logical to have some global config to config this, but it is not something that should be done in this PR. So current approach looks good.